stl/inc/xstring

```diff
@@ -3165,21 +3159,24 @@ public:
                 auto&& _Right_alproxy = _GET_PROXY_ALLOCATOR(_Alty, _Right_al);
                 _Container_proxy_ptr<_Alty> _New_proxy(_Right_alproxy, _Leave_proxy_unbound{}); // throws
 
-                if (_Right._Mypair._Myval2._Large_string_engaged()) {
-                    const auto _New_size     = _Right._Mypair._Myval2._Mysize;
-                    const auto _New_capacity = _Calculate_growth(_New_size, 0, _Right.max_size());
+                const auto _Right_size        = _Right._Mypair._Myval2._Mysize;
+                const _Elem* const _Right_ptr = _Right._Mypair._Myval2._Myptr();
+                if (_Entails_large_string(_Right_size)) {
+                    const auto _New_capacity =
+                        _Calculate_growth(_Right_size, _SMALL_STRING_CAPACITY, _Right.max_size());

```

Comment:
[won't be included in this pr]
If this fix is adopted, the second parameter of `_Calculate_growth(x,y,z)` will be uniformly `_SMALL_STRING_CAPACITY` and further rectification will be possible.
There is also a strength use of `_Calculate_growth(newsize)` in `_Construct_from_iter`. According to impl logic it is performing capacity*=1.5 logic. What about replacing the grow logic here with `_Reallocate_grow_by` or `push_back`?https://github.com/microsoft/STL/blob/40640c6df9754ea031f8df7c720b89714d19df97/stl/inc/xstring#L2712